### PR TITLE
Auth 2

### PR DESCRIPTION
**Pull request by ArielLB**
added functions to handle user access and to restrict the users to enter forbidden zones.
added admin user privileges.

### CODE REVIEW

**Comment by jamiecoe** [comment]
Does this close any issues? If so make sure to put `closes #(id of issue)`

File: BookStore/controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+
+exports.requireSignin = expressJwt({
+    secret: process.env.JWT_SECRET,
+    userProperty: "auth"
+});
+
+exports.isAuth = (req, res, next) => {
+    let user = req.profile && req.auth && req.profile._id == req.auth._id;
```
Comment (jamiecoe) [inline BookStore/controllers/auth.js]: Why are you using `let` for this variable? It is not being mutated. Use `const` instead.

Comment (mcorcias) [reply BookStore/controllers/auth.js]: OK

Comment (jamiecoe) [reply BookStore/controllers/auth.js]: You need to actually change this before I'm going to approve

**Pull request merged**